### PR TITLE
[AAE-3638] - implement priority drodpown widget

### PR DESCRIPTION
**Pull request by SilviuCPopa**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://alfresco.atlassian.net/browse/AAE-3638

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [x] Yes
> - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...
Replace priority input widget with dropdown selection.

**Other information**:

### CODE REVIEW

File: lib/process-services-cloud/src/lib/task/services/task-cloud.service.spec.ts [hunk 1]
```diff
@@ -91,10 +92,12 @@ describe('Task Cloud Service', () => {
     beforeEach(async(() => {
         alfrescoApiMock = new AlfrescoApiServiceMock(new AppConfigService(null), new StorageService());
         identityUserService = TestBed.inject(IdentityUserService);
+        translateService = TestBed.inject(TranslationService);
         spyOn(identityUserService, 'getCurrentUserInfo').and.returnValue(cloudMockUser);
         service = new TaskCloudService(alfrescoApiMock,
```
Comment (DenysVuika) [inline lib/process-services-cloud/src/lib/task/services/task-cloud.service.spec.ts:97]: can you inject this service via TestBed intead of calling ctor manually?

File: lib/process-services-cloud/src/lib/task/services/task-cloud.service.ts [hunk 2]
```diff
@@ -287,7 +288,7 @@ export class TaskCloudService extends BaseCloudService {
     }
 
     get priorities(): any[] {
```
Comment (DenysVuika) [inline lib/process-services-cloud/src/lib/task/services/task-cloud.service.ts:290]: Please introduce an interface and provide proper typing in the return value

**Review by DenysVuika — CHANGES_REQUESTED** [review]

File: lib/process-services-cloud/src/lib/task/services/task-cloud.service.spec.ts [hunk 3]
```diff
@@ -89,13 +90,13 @@ describe('Task Cloud Service', () => {
     });
 
     beforeEach(async(() => {
-        alfrescoApiMock = new AlfrescoApiServiceMock(new AppConfigService(null), new StorageService());
+        // alfrescoApiMock = new AlfrescoApiServiceMock(new AppConfigService(null), new StorageService());
```
Comment (DenysVuika) [inline lib/process-services-cloud/src/lib/task/services/task-cloud.service.spec.ts:93]: commented code, not needed probably

File: lib/process-services-cloud/src/lib/task/start-task/components/start-task-cloud.component.ts [hunk 4]
```diff
@@ -100,6 +100,8 @@ export class StartTaskCloudComponent implements OnInit, OnDestroy {
 
     formKey: string;
 
+    priorities: any;
```
Comment (DenysVuika) [inline lib/process-services-cloud/src/lib/task/start-task/components/start-task-cloud.component.ts:103]: missing type

File: lib/process-services-cloud/src/lib/task/task-list/components/task-list-cloud.component.ts [hunk 5]
```diff
@@ -132,14 +134,17 @@ export class TaskListCloudComponent extends BaseTaskListCloudComponent {
...
+                private taskCloudService: TaskCloudService,
                 appConfigService: AppConfigService,
                 userPreferences: UserPreferencesService) {
         super(appConfigService, userPreferences, TaskListCloudComponent.PRESET_KEY);
     }
 
     load(requestNode: TaskQueryCloudRequestModel) {
         this.isLoading = true;
-        this.taskListCloudService.getTaskByRequest(<TaskQueryCloudRequestModel> requestNode).subscribe(
+        this.taskListCloudService.getTaskByRequest(<TaskQueryCloudRequestModel> requestNode).pipe(
```
Comment (DenysVuika) [inline lib/process-services-cloud/src/lib/task/task-list/components/task-list-cloud.component.ts:145]: you don't need casting if `getTaskByRequest` returns correct result, or fix the `getTaskByRequest` to return it

File: lib/process-services-cloud/src/lib/task/task-list/components/task-list-cloud.component.ts [hunk 6]
```diff
@@ -184,4 +189,22 @@ export class TaskListCloudComponent extends BaseTaskListCloudComponent {
         };
         return new TaskQueryCloudRequestModel(requestNode);
     }
+
+    private replacePriorityValues(tasks: any) {
```
Comment (DenysVuika) [inline lib/process-services-cloud/src/lib/task/task-list/components/task-list-cloud.component.ts:193]: missing type

File: lib/process-services-cloud/src/lib/task/task-list/components/task-list-cloud.component.ts [hunk 7]
```diff
@@ -184,4 +189,22 @@ export class TaskListCloudComponent extends BaseTaskListCloudComponent {
...
+        const entries = tasks.list.entries.map((item: any) => {
```
Comment (DenysVuika) [inline lib/process-services-cloud/src/lib/task/task-list/components/task-list-cloud.component.ts:194]: missing type

**Review by DenysVuika — CHANGES_REQUESTED** [review]

File: lib/testing/src/lib/core/pages/card-view/card-view-select-item.page.ts [hunk 8]
```diff
@@ -39,7 +40,16 @@ export class CardSelectItemPage {
         return this.dropdown.getSelectedOptionText();
     }
 
+    async getReadonlyValue() {
```
Comment (DenysVuika) [inline lib/testing/src/lib/core/pages/card-view/card-view-select-item.page.ts:43]: missing return type

**Review by DenysVuika — APPROVED** [review]

**Review by eromano — APPROVED** [review]

**Pull request merged**